horizon/enforcer/api.py

```diff
@@ -268,6 +269,53 @@ async def _is_allowed(query: BaseSchema, request: Request, policy_package: str):
     return await post_to_opa(request, path, opa_input)
 
 
+async def _is_allowed_gopal(query: BaseSchema, request: Request):
+    headers = transform_headers(request)
+    url = f"{sidecar_config.GOPAL_SERVICE_URL}/v1/authz/check"
+    payload = {"input": query.dict()}
+    exc = None
+    _set_use_debugger(payload)
+    try:
+        logger.info(f"calling GOPAL at '{url}' with input: {payload}")
+        async with aiohttp.ClientSession() as session:
+            async with session.post(
+                url,
+                data=json.dumps(payload["input"]) if payload is not None else None,
+                headers=headers,
+                timeout=sidecar_config.OPA_CLIENT_QUERY_TIMEOUT,
+                raise_for_status=True,
+            ) as opa_response:
+                stats_manager.report_success()
+                return await proxy_response(opa_response)
+    except asyncio.exceptions.TimeoutError:
+        stats_manager.report_failure()
+        exc = HTTPException(
+            status.HTTP_504_GATEWAY_TIMEOUT,
+            detail="OPA request timed out (url: {url}, timeout: {timeout}s)".format(
+                url=url,
+                timeout=sidecar_config.OPA_CLIENT_QUERY_TIMEOUT,
+            ),
+        )
+    except aiohttp.ClientResponseError as e:
+        stats_manager.report_failure()
+        exc = HTTPException(
+            status.HTTP_502_BAD_GATEWAY,  # 502 indicates server got an error from another server
+            detail="OPA request failed (url: {url}, status: {status}, message: {message})".format(
```

Comment:
same here: authz service request failed

Reply:
gopal calls opa, there is no way to distinguish between the two